### PR TITLE
add check for number strings

### PR DESCRIPTION
**Pull request by CarolynWebster**
We recently tied mobile builds to our npm library, which means it actually installed the correct version, and likely started catching the issue where number inputs are coming through as strings, ("1000" instead of 1000), and it was failing validation, with an error "Invalid value "200" supplied to /Length". We also see this error when adding hours to labor, since in our schemas those are numbers (most likely to allow decimals like 8.5 hours).

Rainforest errors:
[Broken Length](https://app.rainforestqa.com/runs/555186/tests/167274/browsers/android_phone_71_720_1280/steps/49303106)
[Broken Labor](https://app.rainforestqa.com/runs/555186/tests/167276/browsers/android_phone_71_720_1280/steps/49303096)

We have varied use of the "Number" vs "Integer" in our schemas. My guess is that Integers are expected to always be whole numbers, where Number can be an integer or a float.

Basically adding in a similar util for Number that exists for Integer, but instead of checking if it is a whole number, we just verify the value is a number. I loaded this up locally and it seems to do the trick. I also replaced the library locally for the web app and that also seemed to work fine, but in all fairness the web app is properly passing numbers back, not strings, so likely why we didn't see the issue on the web before now.

### CODE REVIEW

File: src/util.js [hunk 1]
```diff
@@ -6,11 +6,19 @@ function isInteger(n) {
...
-var Int = t.irreducible('Int', isInteger);
+function isNumber(n) {
+	return !isNaN(n) 
+}
 
+var Null = t.irreducible('Null', function (x) {
+  return x === null;
+});
+var Int = t.irreducible('Int', isInteger);
+var Num = t.irreducible('Num', isNumber)
```
Comment (jongoncalves) [inline src/util.js:17]: Is it possible to use JS's Number class to check if it's a number (and integer as well actually)?

Seems  like its interface supports these use cases and would really simplify this code.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number

Comment (CarolynWebster) [reply src/util.js:17]: I think the method has to resolve to True or False, and for Integer, I think we would want to make sure it is a whole number, but I could make it `Number(n) !== NaN` for Number if that seems clearer? `Number.isInteger()` only works if it is a number, it doesn't seem to handle strings, where the current modulo function handles both.

Comment (jongoncalves) [reply src/util.js:17]: I meant the Number class built-in methods:
[Number.isInteger](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger)
[Number.isNan](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN)

They both return bools - so something like:

```javascript
var Int = t.irreducible('Int', Number.isInteger)
var Num = t.irreducible('Num', Number.isNan)
```

Comment (CarolynWebster) [reply src/util.js:17]: I added the last bit to my comment after I tested those out. `Number.isNaN` works with strings or numbers, but `Number.isInteger('3')` results in `false`. But I think using `Number.isNaN` is def an improvement for Number.

Comment (CarolynWebster) [reply src/util.js]: could do something like `Number.isInteger(parseInt(n))` for Integer

Comment (CarolynWebster) [reply src/util.js:17]: Unfortunately - it goes back to failing when I sub out the method, even though in a node shell it lets me give a string to `Number.isNaN` in this use case it doesn't seem to work. So I don't think we can use it.

Comment (jongoncalves) [reply src/util.js:17]: Hummm, that's really weird. Sounds good though.

**Review by jongoncalves — APPROVED** [review]
